core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -259,21 +255,14 @@ static void validateManifestEntries(ManifestFile manifest,
                                       Iterator<Long> ids,
                                       Iterator<DataFile> expectedFiles,
                                       Iterator<ManifestEntry.Status> expectedStatuses) {
-    validateManifestEntries(manifest.path(), ids, expectedFiles, expectedStatuses);
-  }
-
-  static void validateManifestEntries(String manifest,
-                                      Iterator<Long> ids,
-                                      Iterator<DataFile> expectedFiles,
-                                      Iterator<ManifestEntry.Status> expectedStatuses) {
-    for (ManifestEntry entry : ManifestReader.read(localInput(manifest)).entries()) {
+    for (ManifestEntry entry : ManifestReader.read(manifest, FILE_IO).entries()) {
       DataFile file = entry.file();
       DataFile expected = expectedFiles.next();
       final ManifestEntry.Status expectedStatus = expectedStatuses.next();
       Assert.assertEquals("Path should match expected",
           expected.path().toString(), file.path().toString());
       Assert.assertEquals("Snapshot ID should match expected ID",
-          (long) ids.next(), entry.snapshotId());
+          (long) ids.next(), (long) entry.snapshotId());
```

Comment:
Nit: should we get rid of the casts instead of adding a second to `long`? I think the problem is when Java doesn't know whether to use `assertEquals(long, long)` or `assertEquals(Object, Object)` because one is a primitive and the other is boxed. If both are boxed, we should be okay?

Reply:
Done